auth/api/iam/pkce.go

```diff
@@ -0,0 +1,33 @@
+package iam
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"io"
+)
+
+type PKCEParams struct {
+	Challenge       string
+	ChallengeMethod string
+	Verifier        string
+}
+
+func randomString(n int) string {
```

Comment:
`crypto.SecureRandom()` is used everywhere else

Reply:
I cannot find `crypto.SecureRandom() ` but instead a method named `crypto.GenerateNonce()` looks a lot like it. Used it instead.